src/lobster/data/_ume_datamodule.py

```diff
@@ -0,0 +1,188 @@
+
+SUPPORTED_DATASETS = {"M320M", "ChEMBL", "Calm", "Uniref50"}
+
+
+class UmeLightningDataModule(LightningDataModule):
+    def __init__(
+        self,
+        *,
+        root: Path | str,
+        tokenizer_max_length: int,
+        datasets: None | Sequence[str] | Dict[str, float] = None,
+        download: bool = False,
+        use_text_descriptions: bool = False,
+        generator: Generator | None = None,
+        seed: int = 0xDEADBEEF,
+        batch_size: int = 1,
+        num_workers: int = 0,
+        collate_fn: Callable[[list[T]], Any] | None = None,
+        max_length: int = 512,
+        pin_memory: bool = True,
+        drop_last: bool = False,
+        shuffle: bool = True,
+        length: Sequence[float] = (0.9, 0.05, 0.05),
+    ) -> None:
+        super().__init__()
+
+        if isinstance(datasets, dict):
+            self.dataset_names = list(datasets.keys())
+            self.weights = list(datasets.values())
+        else:
+            self.dataset_names = list(datasets) if datasets is not None else list(SUPPORTED_DATASETS)
+            self.weights = [1.0] * len(self.dataset_names)
+
+        if not set(self.dataset_names).issubset(SUPPORTED_DATASETS):
+            raise ValueError(
+                f"Only the following datasets are supported: {SUPPORTED_DATASETS}. "
+                f"Unknown datasets: {set(self.dataset_names) - SUPPORTED_DATASETS}"
+            )
+
+        self._root = root
+        self._download = download
+        self._tokenizer_max_length = tokenizer_max_length
+        self._use_text_descriptions = use_text_descriptions
+        self._generator = generator or Generator().manual_seed(seed)
+        self._seed = seed
+        self._batch_size = batch_size
+        self._max_length = max_length
+        self._num_workers = num_workers
+        self._collate_fn = collate_fn
+        self._pin_memory = pin_memory
+        self._drop_last = drop_last
+        self._shuffle = shuffle
+        self._lengths = length
+
+        self._datasets = None
+        self._train_datasets = None
+        self._val_datasets = None
+        self._test_datasets = None
+
+    def _get_tokenizer_transform(self, modality: Literal["SMILES", "amino_acid", "nucleotide"]) -> None:
+        match modality:
+            case "SMILES":
+                tokenizer = SmilesTokenizerFast()
+            case "amino_acid":
+                tokenizer = AminoAcidTokenizerFast()
+            case "nucleotide":
+                tokenizer = NucleotideTokenizerFast()
+
+        return TokenizerTransform(
+            tokenizer, padding="max_length", truncation=True, max_length=self._tokenizer_max_length
+        )
+
+    def prepare_data(self) -> None:
+        smiles_transform = self._get_tokenizer_transform("SMILES")
+        amino_acid_transform = self._get_tokenizer_transform("amino_acid")
+        nucleotide_transform = self._get_tokenizer_transform("nucleotide")
+
+        self._datasets = []
+
+        # SMILES
```

Comment:
this is good for now, but maybe add a TODO to handle this with a method that takes a dataset name and returns the corresponding dataset